omniscidb/QueryEngine/CostModel/DataSources/DwarfBench.cpp

```diff
@@ -13,69 +13,41 @@
 
 #include "DwarfBench.h"
 
+#include "Logger/Logger.h"
+
+#include <bench.hpp>
+
 #include <fstream>
 #include <iostream>
 
 namespace costmodel {
 
-DwarfBenchDataSource::DwarfBenchDataSource()
-    : DataSource(DataSourceConfig{.dataSourceName = "DwarfBench",
-                                  .supportedDevices = {ExecutorDeviceType::CPU},
-                                  .supportedTemplates = {AnalyticalTemplate::GroupBy,
-                                                         AnalyticalTemplate::Join,
-                                                         AnalyticalTemplate::Scan}}) {}
+struct DwarfBenchDataSource::PrivateImpl {
+  DwarfBench::Dwarf convertToDwarf(AnalyticalTemplate templ);
+  DwarfBench::DeviceType convertDeviceType(ExecutorDeviceType device);
+  std::vector<Detail::Measurement> convertMeasurement(
+      const std::vector<DwarfBench::Measurement> measurements);
```

Comment:
What's the reason you're passing the vector by value?

Reply:
oops =)